src/AutoRest.CSharp/Common/Generation/Writers/JsonCodeWriterExtensions.cs

```diff
@@ -217,7 +218,14 @@ public static void ToSerializeCall(this CodeWriter writer, JsonSerialization ser
                                 var systemObjectType = objectType as SystemObjectType;
                                 if (systemObjectType != null && IsCustomJsonConverterAdded(systemObjectType))
                                 {
-                                    writer.Append($"JsonSerializer.Serialize(writer, {name});");
+                                    var optionalSerializeOptions = string.Empty;
+                                    if (systemObjectType.SystemType == typeof(ManagedServiceIdentity) && valueSerialization.Version == "v3")
+                                    {
+                                        writer.UseNamespace("Azure.ResourceManager.Models");
+                                        writer.Line($"var serializeOptions = new JsonSerializerOptions {{ Converters = {{ new {nameof(ManagedServiceIdentityTypeV3Converter)}() }} }};");
```

Comment:
This could work, but the generated code becomes:
`var serializeOptions = new JsonSerializerOptions { Converters = { new ResourceManager.Models.ManagedServiceIdentityTypeV3Converter() } };`

`Azure.ResourceManager.Models` is added automatically, but not sure why it's using `ResourceManager.Models.ManagedServiceIdentityTypeV3Converter`.